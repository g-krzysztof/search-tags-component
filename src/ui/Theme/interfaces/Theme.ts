import {
  Borders,
  Colors,
  FontSizes,
  FontWeights,
  LineHeights,
  Motion,
  Radii,
  Space,
} from '.'

export interface Theme {
  borders: Borders
  colors: Colors
  fontSizes: FontSizes
  fontWeights: FontWeights
  lineHeights: LineHeights
  motion: Motion
  radii: Radii
  space: Space
}
