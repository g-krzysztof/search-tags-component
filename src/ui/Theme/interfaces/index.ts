export type { Borders } from './Borders'
export type { Colors } from './Colors'
export type { FontSizes } from './FontSizes'
export type { FontWeights } from './FontWeights'
export type { LineHeights } from './LineHeights'
export type { Motion } from './Motion'
export type { Radii } from './Radii'
export type { Space } from './Space'
export type { Theme } from './Theme'
