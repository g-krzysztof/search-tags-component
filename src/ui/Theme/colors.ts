import { Colors } from './interfaces'

export const colors: Colors = {
  blue: '#2463ec',
  redDark: '#c22f3f',
  redLight: '#ff686f',
  greenDark: '#3b8b58',
  greenLight: '#1ea24f',
  purple: '#a461e5',
  grayLight: '#f4f4f6',
  grayMedium: '#a5a4a9',
  grayDark: '#626264',
  black: '#161616',
  white: '#ffffff',
}
