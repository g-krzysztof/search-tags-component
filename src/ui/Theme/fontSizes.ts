import { FontSizes } from './interfaces'

// values should be interpreted as rem
// rem base: 1rem = 16px

export const fontSizes: FontSizes = {
  xs: 10,
  s: 12,
  m: 14,
  l: 16,
  xl: 18,
  xxl: 32,
}
