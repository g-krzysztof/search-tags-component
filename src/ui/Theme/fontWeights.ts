import { FontWeights } from './interfaces'

export const fontWeights: FontWeights = {
  regular: 400,
  medium: 500,
  bold: 700,
}
