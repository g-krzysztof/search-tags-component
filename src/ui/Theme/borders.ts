import { Borders } from './interfaces'
import { colors } from './colors'

export const borders: Borders = {
  // solid
  grayLight: `2px solid ${colors.grayLight}`,
  grayLightThin: `1px solid ${colors.grayLight}`,
  grayMedium: `2px solid ${colors.grayMedium}`,
  grayMediumThin: `1px solid ${colors.grayMedium}`,
  blueMedium: `2px solid ${colors.blue}`,
  whiteMedium: `2px solid ${colors.white}`,
}
