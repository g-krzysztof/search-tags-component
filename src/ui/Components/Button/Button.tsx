import { Box } from '../Box'
import { Text } from '../Text'
import { Colors } from '../../Theme/interfaces'
import styled from 'styled-components'

export interface ButtonProps {
  color?: keyof Colors
  active?: boolean
  onClick?: () => void
  variant?: 'default' | 'bare'
  label: string
}

export const Button: React.FC<ButtonProps> = ({
  active = false,
  onClick,
  label,
}) => {
  return (
    <StyledBox
      display="flex"
      justifyContent="center"
      alignItems="center"
      borderRadius="rounded"
      active={active}
      onClick={onClick && onClick}
      py="xxs"
    >
      <Text color={active ? 'white' : 'black'} fontSize="s" fontWeight="medium">
        {label}
      </Text>
    </StyledBox>
  )
}

const StyledBox = styled(Box)<{
  active: boolean
  onClick?: () => void
}>`
  cursor: ${({ active }) => (active ? 'pointer' : 'not-allowed')};
  transition: 0.2s;
  background-color: ${({ theme: { colors }, active }) =>
    active ? colors.blue : colors.grayLight};
  &:hover {
    background-color: ${({ theme: { colors }, active }) =>
      active ? colors.blueDark : colors.grayLight};
  }
`
