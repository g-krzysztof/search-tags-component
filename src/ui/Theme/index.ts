import { borders } from './borders'
import { colors } from './colors'
import { fontSizes } from './fontSizes'
import { fontWeights } from './fontWeights'
import { Theme } from './interfaces'
import { lineHeights } from './lineHeights'
import { motion } from './motion'
import { radii } from './radii'
import { space } from './space'

export { borders } from './borders'
export { colors } from './colors'
export { fontSizes } from './fontSizes'
export { fontWeights } from './fontWeights'
export type { Theme } from './interfaces'
export { lineHeights } from './lineHeights'
export { motion } from './motion'
export { radii } from './radii'
export { space } from './space'

export const theme: Theme = {
  borders,
  colors,
  fontSizes,
  fontWeights,
  lineHeights,
  motion,
  radii,
  space,
}
