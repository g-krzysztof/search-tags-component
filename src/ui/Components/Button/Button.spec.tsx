import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'
import { ThemeProvider } from 'styled-components'
import { theme } from '../../Theme'
import { Button } from './Button'

describe('Button', () => {
  const mockOnClick = jest.fn()

  it('renders the button with the provided label', () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Button label="Test Button" onClick={mockOnClick} />
      </ThemeProvider>,
    )

    expect(getByText('Test Button')).toBeInTheDocument()
  })

  it('calls the onClick function when the button is clicked and active', () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Button label="Test Button" onClick={mockOnClick} active />
      </ThemeProvider>,
    )

    fireEvent.click(getByText('Test Button'))
    expect(mockOnClick).toHaveBeenCalled()
  })

  it('renders the button with the correct text color when active', () => {
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Button label="Test Button" onClick={mockOnClick} active color="blue" />
      </ThemeProvider>,
    )

    expect(getByText('Test Button')).toHaveStyle('color: rgb(255, 255, 255)')
  })

  it('renders Button unchanged', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button label="Test Button" onClick={mockOnClick} />
      </ThemeProvider>,
    )
    expect(container).toMatchSnapshot()
  })
})
