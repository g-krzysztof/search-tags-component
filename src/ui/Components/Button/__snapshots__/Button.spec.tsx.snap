// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button unchanged 1`] = `
<div>
  <div
    alignitems="center"
    borderradius="rounded"
    class="sc-8b3de090-0 pfEju sc-356cdfa7-0 fXPOIS"
    display="flex"
    justifycontent="center"
    py="xxs"
  >
    <div
      class="sc-5a158f7a-0 fQHTZO"
      color="black"
      font-size="s"
      font-weight="medium"
    >
      Test Button
    </div>
  </div>
</div>
`;
